plugin/autoload.vim

```diff
@@ -0,0 +1,2 @@
+exe "lua require('textcase.plugin.presets').Initialize()"
```

Comment:
@johmsalas so this file can be lua it doesn't need to be a vim file. Also, just to clarify this will run for all users so if the point is to avoid calling initialize for all users it might be better to add a flag to the config and only run initialize if the flag is true or false in the setup function.

Also FWIW I think autoload might not be a good name for this file (it's not a real problem) but nvim has the concept of an autoload directory (see `:h autoload`) already so if you call the file autoload contributors might think it's related, but it won't be.

Reply:
> this will run for all users so if the point is to avoid calling initialize for all users

Initialize should run for all the users, even if the setup method is not called. Other users want the plugin behavior but, not default mappings, that's all

> this file can be lua it doesn't need to be a vim file

It seems like [LUA code is not loaded by default](https://www.linode.com/docs/guides/writing-a-neovim-plugin-with-lua/), it should be invoked by vimscript. That's this new file, a default invoker. I'd seen this pieces of code in other plugins, for instance [Telescope](https://github.com/nvim-telescope/telescope.nvim/blob/master/autoload/health/telescope.vim), and that's probably why. At it is also [in this docs](https://www.linode.com/docs/guides/writing-a-neovim-plugin-with-lua/), they build the plugin using LUA, and then initialize it with a couple of lines in vimscript

TIL about autoload. yeah, it could be confusing, will change the name. Thanks for your guidance

Reply:
@johmsalas let me clarify since this can all be confusing. As you say, lua code is not loaded by default, that is correct. Using a `plugin/` dir to load it is valid. What I'm saying is that you can just as well put a `lua` file in the `plugin` dir as well as `vim` file. It doesn't have to be `vim` it doesn't really matter, but just letting you know. The reason you see people using `.vim` in the plugin dir is that those plugins were made before this functionality was added in `0.6` or so. Try it locally and see. I use it in my dotfiles as well as in some of my plugins.